### PR TITLE
replace methods taking (buf, off, len) with ones taking RawPacket

### PR DESCRIPTION
**Pull request by bbaldino**

### CODE REVIEW

File: src/org/jitsi/impl/neomedia/MediaStreamImpl.java [hunk 1]
```diff
@@ -4028,31 +4045,44 @@ public REDBlock getPrimaryREDBlock(ByteArrayBuffer baf)
...
+
+            return REDBlockIterator.getPrimaryBlock(pkt.getBuffer(), pkt.getOffset(), pkt.getLength());
         }
         else
         {
-            final int payloadOff = RawPacket.getPayloadOffset(buf, off, len),
-                payloadLen = RawPacket.getPayloadLength(buf, off, len, true);
-
-            return new REDBlock(buf, payloadOff, payloadLen, pktPT);
+            return new REDBlock(pkt.getBuffer(), pkt.getPayloadOffset(), pkt.getPayloadLength(), pktPT);
```
Comment (bgrozev) [inline src/org/jitsi/impl/neomedia/MediaStreamImpl.java]: Line length

File: src/org/jitsi/impl/neomedia/codec/REDBlockIterator.java [hunk 2]
```diff
@@ -16,6 +16,8 @@
 package org.jitsi.impl.neomedia.codec;
 
 import java.util.*;
+
+import org.jitsi.service.neomedia.*;
```
Comment (bgrozev) [inline src/org/jitsi/impl/neomedia/codec/REDBlockIterator.java]: Why is this necessary?

File: src/org/jitsi/impl/neomedia/rtp/MediaStreamTrackDesc.java [hunk 3]
```diff
@@ -413,16 +413,31 @@ RTPEncodingDesc findRTPEncodingDesc(long ssrc)
...
      * specified in the buffer passed in as an argument, or null.
+     * @Deprecated use findFrameDesc(long, long)
      */
+    @Deprecated
     public FrameDesc findFrameDesc(byte[] buf, int off, int len)
     {
-        RTPEncodingDesc rtpEncoding = findRTPEncodingDesc(buf, off, len);
-        if (rtpEncoding == null)
+        RawPacket pkt = new RawPacket(buf, off, len);
+        return findFrameDesc(pkt.getSSRCAsLong(), pkt.getTimestamp());
```
Comment (bgrozev) [inline src/org/jitsi/impl/neomedia/rtp/MediaStreamTrackDesc.java]: We could use the static versions of getSSRCAsLong and getTimestamp here

File: src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java [hunk 4]
```diff
@@ -718,6 +718,23 @@ FrameDesc findFrameDesc(byte[] buf, int off, int len)
...
     /**
+     * Finds the {@link FrameDesc} that matches the RTP packet specified
+     * in the buffer passed in as an argument.
+     *
+     * @param timestamp the timestamp of the desired {@link FrameDesc}
+     *
+     * @return the {@link FrameDesc} that matches the RTP timestamp given,
+     * or null if there is no matching frame {@link FrameDesc}.
+     */
+    FrameDesc findFrameDesc(long timestamp)
```
Comment (bgrozev) [inline src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java]: Why do we keep the findFrameDesc(byte[] buf, int off, int len) method?

**Comment by bgrozev** [comment]
Minor comments left inline. One more question: how do we handle the MediaStream interface?

File: src/org/jitsi/impl/neomedia/MediaStreamImpl.java (hunk 1 above)
Comment (bbaldino) [inline src/org/jitsi/impl/neomedia/MediaStreamImpl.java]: fixed

File: src/org/jitsi/impl/neomedia/codec/REDBlockIterator.java (hunk 2 above)
Comment (bbaldino) [inline src/org/jitsi/impl/neomedia/codec/REDBlockIterator.java]: it's not, not sure how that got in there.  fixed.

File: src/org/jitsi/impl/neomedia/rtp/MediaStreamTrackDesc.java (hunk 3 above)
Comment (bbaldino) [inline src/org/jitsi/impl/neomedia/rtp/MediaStreamTrackDesc.java]: fixed

File: src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java (hunk 4 above)
Comment (bbaldino) [inline src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java]: i just hadn't replaced the existing chain, will remove it.

**Comment by bbaldino** [comment]
it looks like jvb doesn't use the interface type anywhere, only MediaStreamImpl directly, that's why i didn't see any failures even though i didn't add it.  but added it to the interface now
